docs/reference/remap.cue

```diff
@@ -52,17 +51,589 @@ remap: {
+				`%a %d %b %T %z %Y`| [`date`](\(urls.date)) command output with numeric time zone | `Tue 01 Dec 02:37:54 -0700 2020`
+				`%a %d %b %T %#z %Y` | [`date`](\(urls.date)) command output with numeric time zone (minutes can be missing or present) | `Tue 01 Dec 02:37:54 -07 2020`
+				"""
+			use: ["parameter", "return"]
+			examples: [
+				"%a %d %b %T %Y",
+				"%FT%TZ",
+			]
+		}
+	}
+
+	// TRL syntax
+	#Operators: [_category=string]: [_op=string]: string
+
+	syntax: [RuleName=string]: {
+		#InNOut: {
+			in:  string
+			out: #RemapReturnTypes
+		}
+
+		name:        RuleName
+		href:        string // Ensures that we don't end up with clashing anchors
+		description: string
+		examples?: [string, ...string]
+		in_n_out?: [#InNOut, ...#InNOut]
+		operators?: #Operators
+		warnings?: [string, ...string]
+	}
+
+	syntax: {
+		"Paths": {
+			href: "paths"
+
+			description: """
+				In TRL, a dot (`.`) holds state across the script. At the beginning of the script,
+				it represents the object arriving into the transform; that object can be a log or a
+				metric. To give an example, imagine you're writing a TRL script to handle logs in
+				[JSON](\(urls.json)) format. Here's an example event:
+
+				```json
+				{"status_code":200,"username":"booper1234","message":"Successful transaction"}
+				```
+
+				In this case, the event object, represented by the dot, has three fields:
+				`.status_code`, `.username`, and `.message`. You can assign new values to the
+				existing fields (`.message = "something different"`), add new fields (`.new_field =
+				"new value"`), delete fields (`del(.username)`), store those values in variables
+				(`$code = .status_code`), and more.
+
+				### Nested values
+
+				The dot syntax can represent nested fields, for example `.transaction.id` or
+				`.geo.latitude`. If you assign values to nested fields whose parent fields don't
+				exist, the parent fields are created. Take this expression as an example:
+
+				```
+				.user.info.hobbies = ["cooking", "Pogs"]
+				```
+
+				If the `user` field doesn't exist, it is created; if `.user` exists but `.user.info`
+				doesn't, `.user.info` is created; and so on.
+
+				### Path coalescing
+
+				Path *coalescing* in TRL enables you to express "or" logic inside of paths. This
+				expression sets `user.first_name` to `"Feldman"` *if* that field exists; if not, the
+				`user.last_name` field is set to `"Feldman"` instead.
+
+				```
+				.user.(first_name | last_name) = "Feldman"
+				```
+
+				### Quoted paths
+
+				In the examples above, all paths have used literals à la `.foo.bar`. But path
+				segments can also be quoted, as in this example:
+
+				```
+				user.preferences."favorite color" = "chartreuse"
+				```
+
+				Quoted paths are particularly useful when keys need to contain whitespace.
+
+				### Indexing
+
+				Values inside TRL arrays can be accessed via index (beginning with 0). For the array
```

Comment:
I wouldn't call it "TRL arrays" specifically, and probably a good idea to mention that this also works for paths, not just variables:


```rust
.foo[2]
```